### PR TITLE
[ISV-5091] Optimize container tool storage configuration

### PR DESCRIPTION
**Pull request by mavaras**
**TLDR**: specifying a different storage driver (overlay) from the default one (VFS) fixed the issue as it is more optimized for storage.

- Adds a new Ansible step controlled by the boolean variable `optimize_build_storage` to apply a storage configuration that saves disk space and make podman build execution safer. Take into account that the podman instance used here is NOT the same as the one we handle in community-operators-pipeline. The only way to customize it was to edit it straight into the Ansible config that invokes it.
- Adds extra parameters variable to podman build command. This allows to easily add new configuration options and for debugging just editing [this list](https://github.com/redhat-openshift-ecosystem/community-operators-pipeline/blob/e1de0e6d6ebddc0f4fc6f7f4cec2a6f6cd57c926/ci/scripts/opp.sh#L45). The real fix to the issue is to setup overlay as storage driver with `--storage-driver`.

Example start-to-end pipeline run with this changes being applied successfully: https://github.com/k8s-operatorhub/community-operators/actions/runs/10237309646/job/28322358625

Some references:
- https://github.com/containers/buildah/issues/4423
- https://github.com/containers/buildah/issues/2850

### CODE REVIEW

**Review by Allda — APPROVED** [review]

**Review by mporrato — APPROVED** [review]

**Comment by acornett21** [comment]
Is the plan after this is merged to re-run every failed job (in order), and also some jobs that show as passed, but the ohio job didn't run?

**Comment by acornett21** [comment]
What about jobs like this that show green but the entire pipeline never ran? Is there a way to track these and make sure they are also re-ran?

- https://github.com/k8s-operatorhub/community-operators/actions/runs/10221231385

**Comment by mavaras** [comment]
> What about jobs like this that show green but the entire pipeline never ran? Is there a way to track these and make sure they are also re-ran?
> 
> * https://github.com/k8s-operatorhub/community-operators/actions/runs/10221231385

Yes, I'll be checking all of these since the issue started.
